src/modules/exchange/exchangeProviders/exchange-rate.service.ts

```diff
@@ -0,0 +1,18 @@
+import { BaseExchangeService, ExchangeRateData } from '../echange.abstract';
+import { RateResponce } from '../dto/exchange.dto';
+
+export class ExchangeRateService extends BaseExchangeService {
+  private exchangeRateAPIKey = process.env.EXCHANGE_API_KEY;
+
+  requestURL = `https://v6.exchangerate-api.com/v6/${this.exchangeRateAPIKey}/latest/UAH`;
```

Comment:
Why not in the constructor? 🤔 And then to make requestURL private

Reply:
I decided to make requestURL protected to be able to declare it in the parent class and initialize in the children classes

Reply:
I get it but the question is still  stays: why not in the constructor? Just call super(newUrl) and that way you would initialize it